### DISCUSSION TITLE
Setting up plasma-\beta in the mid-plane in 'gr_torus'!

### DISCUSSION THREAD

**Discussion by R-0-n-Y** [PrincetonUniversity/athena#448] [Ideas]
I was trying to run some test problems with "gr_torus". I was wondering whether it is possible to set some profile of plasma-\beta at the midplane of the disk? In the problem segment of the input file "athinput.fm_torus", I find something like \beta-min, does that allow me to set the mid-plane profile to any desired value?

**Comment by c-white** [comment]
First, `problem/beta_min` (which is distinct from `hydro/beta_min`) was removed from the code about 3 years ago, so if possible it would be good to update to the current version.

The issue with that parameter was that it was always approximate. Indeed, predicting the profile of beta for a given field configuration is tricky. Really the thing to do is run a problem for 0 timesteps and analyze the initial conditions that are dumped to file to see if they suit your purposes.

Magnetic fields must be derived from a vector potential, lest they include monopoles, and the `gr_torus.cpp` file has a couple pre-defined, parameterized vector potentials it can use. Without more information, a profile of beta is insufficient to define the field, and so the problem becomes "what vector potential gives a good profile in beta?"

Aside from the simplest geometries (a purely vertical field everywhere, or a purely toroidal field), the initial profile in beta won't even be particularly smooth or monotonic. For example, if there are field loops on the midplane, beta goes to infinity at their centers.

I'll maybe add some advice from my experience: For most applications, the thing that matters is field topology, not strength. Stretching and reconnection of the field will set its amplitude naturally over the course of the simulation, and it would be worrisome if the final state depended on the initial profile. However, topology is much harder to change over a simulation time. Thus pure vertical fields, large poloidal loops, small poloidal loops, and toroidal fields can lead to very different outcomes. So maybe the best question to ask is what field topology do you want.
